sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormat.scala

```diff
@@ -265,8 +265,8 @@ object FileFormat {
    * fields of the [[PartitionedFile]], and do have entries in the file's metadata map.
    */
   val BASE_METADATA_EXTRACTORS: Map[String, PartitionedFile => Any] = Map(
-    FILE_PATH -> { pf: PartitionedFile => pf.toPath.toString },
-    FILE_NAME -> { pf: PartitionedFile => pf.toPath.getName },
+    FILE_PATH -> { pf: PartitionedFile => pf.filePath.urlEncoded },
+    FILE_NAME -> { pf: PartitionedFile => new Path(pf.filePath.urlEncoded).getName },
```

Comment:
is this different from `pf.toPath.getName`?

Reply:
This is interesting, for the same file, `Path` may return different result in `getName` depending on how the `Path` is constructed?

Reply:
Just using `Path.getName` returns the unencoded string (i.e. `some file.txt`).

The new way will return the encoded path name: `some%20file.txt`.

It's not exactly right with URI parameters though... I just updated it to
`new Path(pf.filePath.toUri.getRawPath).getName`